### PR TITLE
Exit org alert added

### LINKED ISSUE #984: [Feature] User is unable to exit an organisation from drawer.
**Describe the bug**
User is unable to exit an organization from drawer.

**To Reproduce**
Visit the drawer in the app.

**Expected behavior**
Organization should get removed from the list and user should exit from same.

**Actual behavior**
The exit org. button is inactive.

**Additional details**
An alert dialogue box could be added before exit to improve UX.

I am willing to work on this issue with discussing features of API for proper functioning with the heads!

### PR DESCRIPTION
**Pull request by bayush-9**
**What kind of change does this PR introduce?**
Alert to the user before exxiting an org

**Issue Number:**

Fixes #984 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**
N.A.

**Summary**
UX improvement.

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Y
Screenshot_1631260280
es

### CODE REVIEW

**Comment by bayush-9** [comment]
The stable version is showing 2.5 which was 2.2.3 before. I think that is why it is showing depreciation warnings ? Can anyone help me with resolving these checks?

**Comment by CyberWake** [comment]
got it @bayush-9. Looking into it. I guess we need to have a new issue opened to fix this deprecations, will respond soon on this.

**Comment by bayush-9** [comment]
@CyberWake  Or let us fix this warnings in this request itself. I may try to replace all warnings ( which it is showing 38 issues right now and create a commit ) using the new v2.5 documentation. What say? After corrections let the code get analyzed and then we will see if it works or not !

**Comment by bayush-9** [comment]
@CyberWake the issue is fixed. The code has a clean run. I made the changes which were demanded by the new version ! There are no major changes though.

**Comment by rutvik11062000** [comment]
@bayush-9 Thank you for pointing this out.
 Please use `  void pushDialog(Widget dialog) {
    showDialog(
        context: navigatorKey.currentContext!,
        barrierColor: Colors.transparent,
        barrierDismissible: false,
        builder: (BuildContext context) {
          return dialog;
        });
  }`
 
  from /services/navigation_service.dart for showing any dialogue, we want to reuse the existing code and add new code only when needed.

File: lib/widgets/custom_drawer.dart [hunk 1]
```diff
@@ -112,37 +113,17 @@ class CustomDrawer extends StatelessWidget {
...
-                                    "Are you sure you want to exit this organization?"),
-                              );
+                        onTap: () => navigationService.pushDialog(
+                          CustomAlertDialog(
+                            reverse: true,
+                            dialogSubTitle:
+                                'Are you sure you want to exit this organization?',
+                            successText: 'Exit',
+                            success: () {
+                              //Exit org
```
Comment (CyberWake) [inline lib/widgets/custom_drawer.dart:123]: It seems like the leave functionality is to be implemented.

**Comment by bayush-9** [comment]
@rutvik11062000, I used locator.dart instead of navigation_screen.dart because context was null in the previous case. Moreover the code is reused now. Your instructions are followed.

File: lib/widgets/custom_drawer.dart (hunk 1 above)
Comment (bayush-9) [reply lib/widgets/custom_drawer.dart:123]: Yes. The work is yet going on at the backend. I mentioned it in the issues but forgot to mention here in the conversation.

Comment (bayush-9) [reply lib/widgets/custom_drawer.dart:123]: https://github.com/PalisadoesFoundation/talawa/issues/984#issuecomment-916703158

**Review by utkarshshendge — COMMENTED** [review]
Hi @bayush-9, thanks for migrating to 2.5.
I noticed that at some places the color is not as expected. See the screenshot.

**Comment by bayush-9** [comment]
@utkarshshendge Sorry for the issue. I would have checked other screens once. The issue is resolved now.
Screenshot_1631418653

**Comment by palisadoes** [comment]
@CyberWake @rutvik11062000 @utkarshshendge Please review.

**Review by rutvik11062000 — APPROVED** [review]
LGTM 👍, But from now I will encourage you to solve / add multiple related bugs in a single PR.

**Comment by palisadoes** [comment]
@utkarshshendge @CyberWake Please review and give your thoughts.

**Review by utkarshshendge — COMMENTED** [review]
@bayush-9 as seen from the screenshot the Appbar color is not reverted back. Added a previous screenshot for reference.

**Comment by bayush-9** [comment]
@utkarshshendge I think its showing in my case. I hope you are talking about the Appbar's color.
Screenshot_1631501226

**Comment by utkarshshendge** [comment]
> @utkarshshendge I think its showing in my case. I hope you are talking about the Appbar's color.

@bayush-9 yes I was talking regarding Appbar's color. It was white for all screens

**Comment by rutvik11062000** [comment]
@bayush-9 Please see work according the style guidelines.
Have a look at mockups: https://www.figma.com/file/GdvI0NPxsIAwQwZ7txfuTF/Talwa-case-study?node-id=0%3A1

**Comment by bayush-9** [comment]
Working on it.

**Comment by bayush-9** [comment]
@utkarshshendge  @rutvik11062000 I am struggling with v2.5 due to depreciations in the themeData's accentcolor properties . Moreover I have fixed the color errors. Any more suggestion on the code are welcomed wherever you think I have overwritten the properties. Thanks for the Figma link, it was very helpful !

**Review by utkarshshendge — DISMISSED** [review]

**Comment by bayush-9** [comment]
@CyberWake @rutvik11062000 Please review the changes.

**Review by rutvik11062000 — APPROVED** [review]

**Comment by bayush-9** [comment]
The floating action (register ) button's color was also having issue in the event details screen. Sorry to collaborators that I am calling too many reviews on a single PR. The issue is fixed now.

**Review by utkarshshendge — APPROVED** [review]

**Comment by bayush-9** [comment]
@rutvik11062000 @CyberWake Please review the changes.

**Review by rutvik11062000 — APPROVED** [review]

**Comment by bayush-9** [comment]
@CyberWake @palisadoes Please merge as many issues are depended on this.

**Pull request merged**